File: Legacy Systems/A4/sieve.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/time.h>

void sieveAlgorithm(int N);

int main()
{
	char input[200] = "";
	int N = 0;	//the inputted upper limit
	while(N == 0)
	{
		printf("Enter upper limit: ");
		scanf("%s", input);
		N = atoi(input);
		if(N == 0)
			printf("Incorrect number! Try again.\n");
	}
	sieveAlgorithm(N);
	return 0;
}

void sieveAlgorithm(int N)
{
	FILE* outputFile = fopen("cOutput.txt.txt", "w");	//the output file for the prime numbers
	int array[N + 1];
	for(int i = 0; i < N; i++)
		array[i] = i;
	
	for(int i = 2; i * i <= N; i++)	//iterating from 2 to the square root of the upper limit
	{
		if(array[i] != 0)
		{
			for(int j = i * 2; j <= N; j += i)	//loop iterates and sets the non-primes to 0
			{
				array[j] = 0;
			}
		}
	}
	for(int i = 2; i < N; i++)	//loop prints the primes to the file
	{
		if(array[i] != 0)
			fprintf(outputFile, "%d\n", array[i]);
	}
	fclose(outputFile);
}

File: Legacy Systems/A4/sieve.py
#!/usr/bin/env/python3
    
if __name__=='__main__':
	N = input("Enter upper limit: ")	#acquiring the upper limit
	outputFile = open("pythonOutput.txt", "w")
	array = []
	for i in range(N+1):
		array.append(i)		#initializing the prime numbers array
		i += 1
	i = 2
	while (i * i <= N):		#loop iterates from 2 to the square root of the upper limit
		if (array[i] != 0):
			for j in range(i * 2, N+1, i):		#loop sets indeces of array that have numbers which are not prime to 0
				array[j] = 0
		i += 1
		
	for i in range(2, N):
		if (array[i] != 0):
			outputFile.write(str(array[i]) + "\n")		#printing the prime numbers to the file
            
	outputFile.close()
